src/App.vue

```diff
@@ -22,15 +22,28 @@ export default {
   data() {
     return {
       direction: '',
+      popStateDetected: false,
     }
   },
   watch: {
     // watch the `$route` to determine the transition to use
     $route(to) {
-      const isListPage = to.name === 'Root' || to.name === 'City'
-      this.direction = isListPage ? 'left' : 'right'
+      // only animate by app navigation
+      if (!this.popStateDetected) {
+        const isListPage = to.name === 'Root' || to.name === 'City'
+        this.direction = isListPage ? 'left' : 'right'
+      } else {
+        this.direction = 'none'
+      }
+      this.popStateDetected = false
     },
   },
+  mounted() {
+    // detect if user is using a back/forward button from the browser
+    window.addEventListener('popstate', () => {
+      this.popStateDetected = true
+    })
```

Comment:
nit: even though this is App and probably the whole tab is being closed, it's still good practice to clean up this listener. Looks like in Vue you use `beforeDestroy` to do that.

Reply:
Nice finding. It's removed now.